pinot-core/src/test/java/org/apache/pinot/core/common/DataFetcherTest.java

```diff
@@ -221,45 +204,23 @@ public void testFetchBigDecimalValues() {
     testFetchBigDecimalValues(FLOAT_COLUMN);
     testFetchBigDecimalValues(DOUBLE_COLUMN);
     testFetchBigDecimalValues(BIG_DECIMAL_COLUMN);
-    testFetchBigDecimalValues(STRING_COLUMN);
     testFetchBigDecimalValues(NO_DICT_INT_COLUMN);
     testFetchBigDecimalValues(NO_DICT_LONG_COLUMN);
     testFetchBigDecimalValues(NO_DICT_FLOAT_COLUMN);
     testFetchBigDecimalValues(NO_DICT_DOUBLE_COLUMN);
     testFetchBigDecimalValues(NO_DICT_BIG_DECIMAL_COLUMN);
-    testFetchBigDecimalValues(NO_DICT_STRING_COLUMN);
   }
 
   @Test
   public void testFetchStringValues() {
-    testFetchStringValues(INT_COLUMN);
-    testFetchStringValues(LONG_COLUMN);
-    testFetchStringValues(FLOAT_COLUMN);
-    testFetchStringValues(DOUBLE_COLUMN);
-    testFetchStringValues(BIG_DECIMAL_COLUMN);
     testFetchStringValues(STRING_COLUMN);
-    testFetchStringValues(NO_DICT_INT_COLUMN);
-    testFetchStringValues(NO_DICT_LONG_COLUMN);
-    testFetchStringValues(NO_DICT_FLOAT_COLUMN);
-    testFetchStringValues(NO_DICT_DOUBLE_COLUMN);
-    testFetchStringValues(NO_DICT_BIG_DECIMAL_COLUMN);
```

Comment:
I understand implicit type cast from STRING to NUMBER is not a good practice. but do we want to support the otherway around?

Reply:
I don't think implicit type cast from other type to STRING is good practice either, and we have observed performance regression doing it in #9287

Reply:
sounds good. good to add release note and backward incompatible tags to explain